lib/puma/server.rb

```diff
@@ -262,6 +259,7 @@ def run(background=true, thread_name: 'srv')
 
       @status = :run
 
+      @options[:puma_server] = self
```

Comment:
Can we not pass this into the thread pool initialiser? It doesn't feel right to mutate the options at this stage.